src/app/components/common/shapes/index.tsx

```diff
@@ -17,74 +17,86 @@ export class ShapesComponent /*extends React.Component<Props, State>*/ implement
 
     private generateName(shapeType: string): string {
         this.nameGenerationCounter ++;
-        return shapeType + this.nameGenerationCounter.toString;
+        return shapeType + this.nameGenerationCounter.toString();
     }
 
     public addRectangle(width: number, height: number): string {
         const name = this.generateName("Rectangle");
-        let obj = new Rectangle(name, width, height);
-        this.shapes[name] = obj;
+        const rect = new Rectangle(name, width, height);
+        this.shapes[name] = rect;
         return name;
     }
 
     public addEllipse(width: number, height: number): string {
         const name = this.generateName("Ellipse");
-        let obj = new Ellipse(name, width, height);
-        this.shapes[name] = obj;
+        const ellipse = new Ellipse(name, width, height);
+        this.shapes[name] = ellipse;
         return name;
     }
 
     public addTriangle(x1: number, y1: number, x2: number, y2: number, x3: number, y3: number): string {
         const name = this.generateName("Triangle");
-        let obj = new Triangle(name, x1,y1,x2,y2,x3,y3);
-        this.shapes[name] = obj;
+        const tri = new Triangle(name, x1,y1,x2,y2,x3,y3);
```

Comment:
I think we should probably avoid short names/abbreviations like this. It's not apparent to me that `tri` means `triangle` without looking at the right hand side of this equal sign which defeats the purpose of renaming these from `obj`. Similarly, `img` for `image` isn't that helpful--and might not be apparent to some people that that is short for image.